### PR TITLE
labelToBlock(): replace unreachable error message with assert

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/s2ir.d [hunk 1]
```diff
@@ -133,10 +133,8 @@ private Label *getLabel(IRState *irs, Blockx *blx, Statement s)
 private block *labelToBlock(IRState *irs, const ref Loc loc, Blockx *blx, LabelDsymbol label, int flag = 0)
 {
     if (!label.statement)
-    {
-        error(loc, "undefined label `%s`", label.toChars());
-        return null;
-    }
+        assert(0);              // should have been caught by GotoStatement.checkLabel()
```
Comment (rainers) [inline src/dmd/s2ir.d]: labelToBlock is also called for inline asm. Are there test cases covering that? It seems this error message is used by
```
void main()
{
    asm
    {
        jmp L1;
    L2:
        nop;
    }
}
```

Comment (WalterBright) [reply src/dmd/s2ir.d]: I put the assert there so we can detect any other cases.

Comment (rainers) [reply src/dmd/s2ir.d]: Not sure I understand. Replacing a reasonable error message with an ICE looks like a regression to me.

**Review by rainers — CHANGES_REQUESTED** [review]
Please add and fix the given test case.

File: src/dmd/s2ir.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/s2ir.d]: you're right

**Comment by WalterBright** [comment]
> Please add and fix the given test case.
Done.

**Comment by WalterBright** [comment]
The changes to root/aav.d are pretty much all indent whitespaces.

File: src/dmd/statement.h [hunk 2]
```diff
@@ -698,6 +698,8 @@ class LabelDsymbol : public Dsymbol
 public:
     LabelStatement *statement;
 
+    bool deleted;               // set if deleted
```
Comment (rainers) [inline src/dmd/statement.h]: It isn't very obvious what "deleted" means in this context. Could you please elaborate in the comment?

**Comment by rainers** [comment]
> Done.

Thanks. Just played around a bit (trying to understand what "deleted" means), here's another test case that still runs into the assert:
```
struct S
{
    static int opApply(int delegate(ref int) dg)
    {
        return 0;
    }
}

void main()
{
    foreach(f; S)
    {
        asm
        {
            jmp L1;
        }
        goto L1;
    }
    L1:;
}
```

**Comment by WalterBright** [comment]
> here's another test case that still runs into the assert:

Devious! fixed

**Comment by WalterBright** [comment]
@rainers this is ready to go!

**Review by rainers — APPROVED** [review]
LGTM

**Comment by RazvanN7** [comment]
This has caused a regression: https://issues.dlang.org/show_bug.cgi?id=20829

**Pull request merged**